src/helia-server.ts

```diff
@@ -272,8 +266,26 @@ export class HeliaServer {
      */
     request.raw.on('close', cleanupFn)
 
+    if (timeout != null) {
+      setTimeout(() => {
```

Comment:
Timeouts like this can keep the process running.  Better to use `AbortSignal.timeout(ms)` which doesn't, and combine signals with `any-signal`.

Reply:
I should have cleared the timeout here, good catch, thanks. any-signal does not prevent duplicate handlers from being added to the same signal, so I prefer not to use it. 

I will open an issue to update this to abortSignal.timeout(50) and `addEventListener`.

Reply:
> any-signal does not prevent duplicate handlers from being added to the same signal, so I prefer not to use it.

Are you passing the same signal to `anySignal` multiple times?  Sounds like a bug if so.

Reply:
I'm not, but I know libraries that do :)